fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -251,16 +252,25 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 	struct Writer : IThreadPoolReceiver {
 		DB& db;
 		UID id;
+		std::shared_ptr<rocksdb::RateLimiter> rateLimiter;
 		Reference<Histogram> commitLatencyHistogram;
 		Reference<Histogram> commitActionHistogram;
 		Reference<Histogram> commitQueueWaitHistogram;
 		Reference<Histogram> writeHistogram;
 		Reference<Histogram> deleteCompactRangeHistogram;
 
 		explicit Writer(DB& db, UID id)
-		  : db(db), id(id), commitLatencyHistogram(Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP,
-		                                                                   ROCKSDB_COMMIT_LATENCY_HISTOGRAM,
-		                                                                   Histogram::Unit::microseconds)),
+		  : db(db), id(id), rateLimiter(SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_BYTES > 0
+		                                    ? rocksdb::NewGenericRateLimiter(
+		                                          SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_BYTES, // rate_bytes_per_sec
+		                                          100 * 1000, // refill_period_us
+		                                          10, // fairness
+		                                          rocksdb::RateLimiter::Mode::kWritesOnly,
+		                                          true /*auto_tuned*/)
```

Comment:
Do we know the impact of `auto_tuned` in FDB? If it is not clear how it affects our throughput, can we make it a server knob?

Reply:
auto_tuned: Enables dynamic adjustment of rate limit within the range [rate_bytes_per_sec / 20, rate_bytes_per_sec] according to the recent demand for background I/O.

Done.